File: [5 kyu] RGB To Hex Conversion/main.py
def rgb(r, g, b):
    def clamp(color):
        return max(0, min(255, color))

    return "{:02X}{:02X}{:02X}".format(clamp(r), clamp(g), clamp(b))


def rgb(r, g, b):
    def clamp(color):
        return max(0, min(255, color))

    return ("{:02X}" * 3).format(clamp(r), clamp(g), clamp(b))


def rgb(r, g, b):
    clamp = lambda color: max(0, min(255, color))

    return ("{:02X}" * 3).format(clamp(r), clamp(g), clamp(b))


def rgb(r, g, b):
    return "".join("{:02X}".format(max(0, min(255, x))) for x in [r, g, b])

File: [5 kyu] RGB To Hex Conversion/main.rs
fn rgb(r: i32, g: i32, b: i32) -> String {
    fn clamp(color: i32) -> i32 {
        0.max(255.min(color))
    }

    format!("{:02X}{:02X}{:02X}", clamp(r), clamp(g), clamp(b))
}

fn rgb(r: i32, g: i32, b: i32) -> String {
    [r, g, b]
        .iter()
        .map(|&color| format!("{:02X}", 0.max(255.min(color))))
        .collect::<String>()
}

fn rgb(r: i32, g: i32, b: i32) -> String {
    [r, g, b]
        .map(|color| format!("{:02X}", 0.max(255.min(color))))
        .join("")
}

fn rgb(r: i32, g: i32, b: i32) -> String {
    [r, g, b]
        .map(|color| format!("{:02X}", color.clamp(0, 255)))
        .join("")
}

use itertools::Itertools;

fn rgb(r: i32, g: i32, b: i32) -> String {
    [r, g, b]
        .iter()
        .map(|&color| format!("{:02X}", 0.max(255.min(color))))
        .join("")
}
